src/editor/contentState/selectionCtrl.js

```diff
@@ -0,0 +1,58 @@
+const selectionCtrl = ContentState => {
+  // Returns the table from the table cell:
+  //   table <-- thead or tbody <-- tr <-- th or td (cell)
+  ContentState.prototype.getTableFromTableCell = function (block) {
+    const table = this.getParent(this.getParent(this.getParent(block)))
+    if (table && table.type !== 'table') {
+      return null
+    }
+    return table
+  }
+
+  ContentState.prototype.tableCellHandler = function (event) {
+    const { start, end } = this.cursor
+    const startBlock = this.getBlock(start.key)
+    const endBlock = this.getBlock(end.key)
+    const { type: startType } = startBlock
+    const { type: endType } = endBlock
+    if (/th|td/.test(startType) && /th|td/.test(endType)) {
+      if (start.key === end.key) {
+        const { text, key } = startBlock
+        this.cursor = {
+          start: { key, offset: 0 },
+          end: { key, offset: text.length }
+        }
+      } else {
+        const startTable = this.getTableFromTableCell(startBlock)
+        const endTable = this.getTableFromTableCell(endBlock)
+        // Check whether both blocks are in the same table.
+        if (!startTable || !endTable) {
+          console.error('No table found or invalid type.')
+          return
+        } else if (startTable.key !== endTable.key) {
+          // Select entire document
+          return
+        }
+        const firstTableBlock = this.firstInDescendant(startTable)
```

Comment:
@fxha Little optimization:

I think `firstTableCell` would be a better variable name, i think `firstTableBlock` is a table block.

`lastTableBlock` is also need to be renamed.